src/app/api/event/[eventId]/routes.ts

```diff
@@ -0,0 +1,31 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { zObjectId } from '@/types/dataModel/base';
+import CMError, { CMErrorResponse, CMErrorType } from '@/utils/cmerror';
+import { zUpdateEventRequest } from '@/types/dataModel/event';
+import { updateEventAction } from '@/server/actions/Event';
+import { UpdateEventRequest } from '@/types/dataModel/event';
+
+export async function PUT(
+  req: NextRequest,
+  {
+    params,
+  }: { params: { eventId: string; eventUpdatesRequest: UpdateEventRequest } }
```

Comment:
params should only have the event Id. the eventUpdateRequest will come from the request data.